packages/react-native-editor/__device-tests__/gutenberg-editor-lists.test.js

```diff
@@ -2,28 +2,38 @@
  * Internal dependencies
  */
 import { blockNames } from './pages/editor-page';
-import { backspace, isAndroid } from './helpers/utils';
+import { backspace, isAndroid, isLocalEnvironment } from './helpers/utils';
 
 describe( 'Gutenberg Editor tests for List block', () => {
 	// Prevent regression of https://github.com/wordpress-mobile/gutenberg-mobile/issues/871
 	it( 'should handle spaces in a list', async () => {
 		await editorPage.addNewBlock( blockNames.list );
-		let listBlockElement = await editorPage.getBlockAtPosition(
-			blockNames.list
-		);
-		// Click List block on Android to force EditText focus
-		if ( isAndroid() ) {
-			await listBlockElement.click();
-		}
+		let listBlockElement = await editorPage.getListBlock();
 
 		// Send the list item text.
-		await editorPage.sendTextToListBlock( listBlockElement, '  a' );
+		await editorPage.typeTextToTextBlock( listBlockElement, '  a', false );
 
 		// Send an Enter.
-		await editorPage.sendTextToListBlock( listBlockElement, '\n' );
+		await editorPage.typeTextToTextBlock( listBlockElement, '\n', false );
+
+		// Click List block on Android.
+		// Only needed when testing in local environment
+		if ( isAndroid() && isLocalEnvironment() ) {
+			await listBlockElement.click();
+		}
 
 		// Send a backspace.
-		await editorPage.sendTextToListBlock( listBlockElement, backspace );
+		await editorPage.typeTextToTextBlock(
+			listBlockElement,
+			backspace,
+			false
+		);
+
+		// There is a delay in Sauce Labs when a key is sent
+		// There isn't an element to check as it's being typed into an element that already exists, workaround is to add this wait until there's a better solution
+		if ( isAndroid() && ! isLocalEnvironment() ) {
+			await editorPage.driver.sleep( 1500 );
+		}
```

Comment:
I'm wondering if we should apply this workaround for other calls to `typeTextToTextBlock`, wdyt?

Reply:
I think it would be better if it's here, as this is the one case I see this happening on, `typeTextToTextBlock` is used quite extensively so having a wait there might increase the test run time.

Reply:
Ok, I see. In that case, should we wait only when running the test in the remote environment? Should we apply this tweak in both?

Reply:
Yup for this and the above lines 21-23, I've made some changes so that it applies to both environments, I created `waitIfAndroid();` for this with a wait time of 1 second for both environments, and looks to be working (latest CI is green and works on local too). This way we can remove the need to introduce `isLocalEnvironment` to this.

Reply:
Resolving this conversion as I see that this comment was addressed in https://github.com/WordPress/gutenberg/pull/40670/commits/4f6917ea29be33315f89b96ea811278408390207.